plugins/woocommerce-blocks/tests/e2e/tests/templates/product-search-results.block_theme.spec.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { test, expect } from '@woocommerce/e2e-utils';
+
+test.describe( 'Product Search Results template', () => {
+	// This is a test to verify there are no regressions on
+	// https://github.com/woocommerce/woocommerce/issues/48489
+	test( 'loads the correct template in the Site Editor', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		const templateName = 'Product Search Results';
+		await admin.visitSiteEditor();
+		await page.getByRole( 'button', { name: 'Templates' } ).click();
+		await page.getByPlaceholder( 'Search' ).fill( templateName );
+		// Wait until search has finished.
+		const searchResults = page.getByLabel( 'Actions' );
+		await expect.poll( async () => await searchResults.count() ).toBe( 1 );
```

Comment:
This is an equivalent to:
```suggestion
		await expect( searchResults ).toHaveCount( 1 );
```

Reply:
Much nicer, thanks! I merged it. :slightly_smiling_face: